internal/tools/pipeline/providers/reconciler/taskrun/taskop/wait_test.go

```diff
@@ -74,3 +78,25 @@ func TestCalculateNextLoopTimeDuration(t *testing.T) {
 		assert.Equal(t, tt[i].want, w.calculateNextLoopTimeDuration(tt[i].loopedTimes).String())
 	}
 }
+
+func Test_waitWhenDone(t *testing.T) {
+	taskRun := &taskrun.TaskRun{
+		Task: &spec.PipelineTask{},
+		P: &spec.Pipeline{
+			PipelineBase: spec.PipelineBase{
+				ID: 1,
+			},
+		},
+	}
+	pm1 := monkey.PatchInstanceMethod(reflect.TypeOf(taskRun), "AppendLastMsg", func(tr *taskrun.TaskRun, msg string) error {
+		return nil
```

Comment:
you should print msgs at least.

Reply:
otherwise, this unit-test cannot guarantee correctness.

Reply:
added